### ISSUE TITLE
Has GitHub abandoned this project?

### ISSUE THREAD

**Issue by VitalyAnkh** [github/semantic#697]
There are no active contributions to this project for more than half a year. As a project backed by a business company, that's abnormal. Just want to know if GitHub decided to abandon this.

**Comment by dcreager** [comment]
Hi @VitalyAnkh, thanks for reaching out.  This question has come up before, e.g. 

- https://github.com/github/semantic/issues/647

It's true that we have not been doing much in this repository recently. As you might know, semantic builds on the [tree-sitter](https://tree-sitter.github.io/tree-sitter/) parsing library, adding more advanced program analysis functionality. Awhile back, we added a [query language](https://tree-sitter.github.io/tree-sitter/using-parsers#pattern-matching-with-queries) to tree-sitter. With that change, there are more Code Productivity features that we can implement purely using tree-sitter, without having to bring in the extra sophistication of semantic. Code Navigation is one of those features, and is what our team has been spending most of our time on for the past couple of years. In particular, we have developed a new [stack graphs][] framework, which is what powers Precise Code Nav on GitHub at the moment.

[stack graphs]: https://github.blog/2021-12-09-introducing-stack-graphs/

So, in short, it's just a question of our team's product focus. Right now we're working on features that can be implemented purely in tree-sitter, and so that's where we're spending most of our time. If/when we move on to product features that require abstract interpretation, or other deep forms of semantic analysis, our focus will shift back.

In the meantime, the repository remains public and open-source, so others are free to continue using it and extending it.
